compiler/rustc_trait_selection/src/error_reporting/traits/on_unimplemented.rs

```diff
@@ -414,7 +414,7 @@ impl<'tcx> OnUnimplementedDirective {
             let cond = item_iter
                 .next()
                 .ok_or_else(|| tcx.dcx().emit_err(EmptyOnClauseInOnUnimplemented { span }))?
-                .meta_item()
+                .meta_item_or_bool()
```

Comment:
This method is used in the first commit but only defined in the second commit, which is a little strange when reading the commits one at a time.